luda/editor/client/src/timeline/renderSubtitleTrackBody.ts

```diff
@@ -0,0 +1,178 @@
+import {
+  Render,
+  Rect,
+  ColorUtil,
+  BorderPosition,
+  MouseButton,
+  Mathu,
+} from "namui";
+import { Clip } from "../type";
+import { renderClip } from "./clip/renderClip";
+import { TimelineState, Track } from "./type";
+
+export const renderSubtitleTrackBody: Render<
+  { timelineState: TimelineState; track: Track },
+  {
+    width: number;
+    height: number;
+  }
+> = (state, props) => {
+  const { clips } = state.track;
+
+  // this should be called before constrainDraggingClipPlacement.
+  const draggingFakeClip = DraggingFakeClip(state, {
+    clips,
+    width: props.width,
+    height: props.height,
+  });
+
+  constrainDraggingClipPlacement(state.timelineState, clips);
+
+  return [
+    Rect({
+      x: 0,
+      y: 0,
+      width: props.width,
+      height: props.height,
+      style: {
+        fill: {
+          color: ColorUtil.Color01(0.4, 0.4, 0.4),
+        },
+        stroke: {
+          color: ColorUtil.Black,
+          width: 1,
+          borderPosition: BorderPosition.inside,
+        },
+      },
+      onMouseUp(event) {
+        if (event.button === MouseButton.right) {
+          const clickMs =
+            state.timelineState.layout.startMs +
+            event.translated.x * state.timelineState.layout.msPerPixel;
+          state.timelineState.contextMenu = {
+            type: "trackBody",
+            clickMs,
+            x: event.x,
+            y: event.y,
+            trackId: state.track.id,
+          };
+        }
+      },
+    }),
+    Rect({
+      x: 0,
+      y: props.height / 3,
+      width: props.width,
+      height: props.height / 3,
+      style: {
+        stroke: {
+          color: ColorUtil.Color01(0.25, 0.25, 0.25),
+          width: 1,
+          borderPosition: BorderPosition.inside,
+        },
+      },
+    }),
+    Rect({
```

Comment:
I think this Rect is duplicated with first Rect. Could you check it?

Reply:
That was hack for redrawing outline. because while drawing middle line, I overwrited outline with middle line color.
Thanks to `BorderPosition.outside`, It has been replaced easily by using `Clip`
https://github.com/NamseEnt/namseent/pull/18/commits/78cea485cd547097168c79ae02e8014de04a2dac